agent/config/parse_windows.go

```diff
@@ -75,8 +75,15 @@ func parseFSxWindowsFileServerCapability() BooleanDefaultFalse {
 		return BooleanDefaultFalse{Value: ExplicitlyDisabled}
 	}
 
+	// By default or if ECS_FSX_WINDOWS_FILE_SERVER_SUPPORTED is set as true, agent will
+	// broadcast the FSx capability.
 	envStatus := utils.ParseBool(os.Getenv("ECS_FSX_WINDOWS_FILE_SERVER_SUPPORTED"), true)
-	return checkDomainJoinWithEnvOverride(envStatus)
+	if envStatus {
+		return BooleanDefaultFalse{Value: ExplicitlyEnabled}
```

Comment:
can we make use of [parseBooleanDefaultTrueConfig()](https://github.com/aws/amazon-ecs-agent/blob/6733da5a6e212a62e16021619f5fcbd3ad879c0a/agent/config/parse.go#L254) to evaluate this? It adds the value based on whether customer explicitly set it or not.

Notice parseBooleanDefaultTrueConfig() and not parseBooleanDefaultFalseConfig() since the default is true.

Reply:
also, in the DefaultConfig, we should change [this](https://github.com/aws/amazon-ecs-agent/blob/2cdd37099090dd976041eb7e9596f157cd6e5d67/agent/config/config_windows.go#L148) to

`FSxWindowsFileServerCapable: BooleanDefaultTrue{Value: ExplicitlyDisabled}`, implying that the default is true unless explicitly disabled?

Reply:
Thanks Anuj!
Addressed all the feedback in the latest revision.